### PR TITLE
[SERF-2971] Instrument AWS clients using `aws-sdk-go` version 2

### PR DESCRIPTION
**Pull request by Neurostep**
## Description

In this PR we introduce a new AWS instrumentation function `InstrumentAWSClient`. It uses [aws-sdk-go-v2](https://github.com/aws/aws-sdk-go-v2). The usage of the `InstrumentAWSSession` marked is deprecated.

Additionally, we updated the `PubSub` configuration to use AWS Config instead of the AWS session to support the migration to AWS SDK Go version 2.

In this PR we keep AWS version 1 SDK to have a smooth transition to a AWS SDK Go version 2.

The usage of the AWS SDK Go version 1 will be cleaned up when we migrate all services.

## Testing considerations

Tested these changes in `user-notifications-ingestion` to confirm the [AWS MSK authentication](https://app.datadoghq.com/apm/traces?query=env%3Adevelopment%20service%3Auser-notifications-ingestion-aws%20operation_name%3ASTS.request%20resource_name%3ASTS.AssumeRole&agg_m=count&agg_m_source=base&agg_t=count&cols=core_service%2Ccore_resource_name%2Clog_duration%2Clog_http.method%2Clog_http.status_code&fromUser=true&graphType=flamegraph&historicalData=true&messageDisplay=inline&query_translation_version=v0&shouldShowLegend=true&sort=time&spanID=5191042327400531903&spanType=all&timeHint=1713888895616&trace=AgAAAY8LvSKA4L0FAAAAAAAAAAAYAAAAAEFZOEx2VE5nQUFEOW0ybExPWjBnd0FwSQAAACQAAAAAMDE4ZjBiYmQtMmQzOS00NzFkLWEyOGMtOGIxOWJlYjE3MThh&traceID=6627de7f00000000480a497e52b9c7bf&traceQuery=&view=spans&start=1713888288505&end=1713889188505&paused=false)

Screenshot 2024-04-23 at 18 18 16

## Checklist

- [x] Prefixed the PR title with the JIRA ticket code 
- [x] Performed simple, atomic commits with [good commit messages][commit messages]
- [x] Verified that the commit history is linear and commits are squashed as necessary
- [x] Thoroughly tested the changes in `development` and/or `staging`
- [x] Updated the `README.md` as necessary

## Related links

* [SERF-2971](https://scribdjira.atlassian.net/browse/SERF-2971)
* [Migration guide](https://aws.github.io/aws-sdk-go-v2/docs/migrating/)
* https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-aws-sdk-for-go-v1-on-july-31-2025/

[commit messages]: https://chris.beams.io/posts/git-commit/

[SERF-2971]: https://scribdjira.atlassian.net/browse/SERF-2971?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by fotos — DISMISSED** [review]
:+1: LGTM 🚀

**Review by laynax — DISMISSED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -1105,29 +1105,38 @@ func main() {
...
+    if err != nil {
+        log.Fatalf("error: %v", err)
+    }
+
+    instrumentation.InstrumentAWSClient(cfg, instrumentation.Settings{
+        AppName: applicationName,
+    })
+
+    // Use the AWS configuration to create clients, such as AWS S3...
+	s3client := s3.NewFromConfig(cfg)
```
[the remark this answers is not in the document]
Comment (Neurostep) [reply README.md]: @fotos Updated the `README` file 👍

**Review by fotos — DISMISSED** [review]
:+1: LGTM :rocket:

**Review by laynax — DISMISSED** [review]

**Review by laynax — APPROVED** [review]

**Review by fotos — APPROVED** [review]
:+1: LGTM 🚀

**Pull request merged**